pages/api/email/[list].ts

```diff
@@ -24,52 +99,30 @@ export default async function handler(
       message: 'Email API: Invalid http method. | Only POST is accepted.',
     });
   }
-  const list = `${req.query.list}`;
-  const email = req.body.email;
 
-  const listId =
-    list === 'session'
-      ? process.env.CAMPAIGN_MONITOR_LIST_SESSION_ID
-      : process.env.CAMPAIGN_MONITOR_LIST_MARKET_RESEARCH_ID;
+  const list: EmailLists = `${req.query.list}` as EmailLists;
+  const email = req.body.email;
 
-  const body: any = {
-    EmailAddress: email,
-    ConsentToTrack: 'Unchanged',
-  };
+  let response;
 
-  if (list !== 'session') {
-    const roles = handleMultiValueSelect(req.body.roles, 'Roles') ?? [];
-    const tags = handleMultiValueSelect(req.body.tags, 'Tags') ?? [];
-
-    body['Name'] = req.body.name ? req.body.name : undefined;
-    body['CustomFields'] = [
-      req.body.country && {
-        Key: 'Country',
-        Value: req.body.country,
-      },
-      ...roles,
-      ...tags,
-    ];
-  }
+  if (list === 'market-research') {
+    response = await makeCMRequest(req);
 
-  const response = await fetch(
-    `https://api.createsend.com/api/v3.2/subscribers/${listId}.json`,
-    {
-      body: JSON.stringify(body),
-      headers: {
-        'Content-Type': 'application/json',
-        Authorization: `Basic ${base64.encode(
-          `${process.env.CAMPAIGN_MONITOR_API_KEY}:x`
-        )}`,
-      },
-      method: 'POST',
+    if (response.status === 201) {
+      res.status(201).json({ email });
+    } else {
+      const result = await response.json();
+      res.status(result.Code).json({ email, message: result.Message });
     }
-  );
-
-  if (response.status === 201) {
-    res.status(201).json({ email });
   } else {
-    const result = await response.json();
-    res.status(result.Code).json({ email, message: result.Message });
+    response = await makeMLRequest(req);
+    // 201 Created: The subscriber was successfully added to the list.
+    // 200 OK: The subscriber was already in the list.
+    if (response.status === 201 || response.status === 200) {
+      res.status(201).json({ email });
+    } else {
+      const result = await response.json();
+      res.status(result.code).json({ email, message: result.error.message });
```

Comment:
According to the api docs errors response bodies look like:
```json
{
  "message": "The given data was invalid.",
  "errors": {
    "email": ["The email must be a valid email address."]
  }
}

```

Reply:
Good catch. Fixed